tests/Module/Carrier/OptionsPageTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Tests\Module\Carrier;
+
+use Packetery\Core\PickupPointProvider\CompoundCarrierCollectionFactory;
+use Packetery\Core\PickupPointProvider\VendorCollectionFactory;
+use Packetery\Latte\Engine;
+use Packetery\Module\Carrier\CarDeliveryConfig;
+use Packetery\Module\Carrier\CountryListingPage;
+use Packetery\Module\Carrier\EntityRepository;
+use Packetery\Module\Carrier\OptionsPage;
+use Packetery\Module\Carrier\PacketaPickupPointsConfig;
+use Packetery\Module\Carrier\WcSettingsConfig;
+use Packetery\Module\FormFactory;
+use Packetery\Module\MessageManager;
+use Packetery\Module\Options\FeatureFlagManager;
+use Packetery\Nette\Http\Request;
+use PHPUnit\Framework\TestCase;
+
+class OptionsPageTest extends TestCase {
+
+	public function setUp(): void {
+		$this->latteEngineMock = $this->createMock(Engine::class);
+		$this->carrierRepositoryMock = $this->createMock(EntityRepository::class);
+		$this->formFactoryMock = $this->createMock(FormFactory::class);
+		$this->request = $this->createMock(Request::class);
+		$this->countryListingPageMock = $this->createMock(CountryListingPage::class);
+		$this->messageManagerMock = $this->createMock( MessageManager::class);
+		$this->featureFlagManagerMock = $this->createMock(FeatureFlagManager::class);
+		$this->carDeliveryConfigMock = $this->createMock(CarDeliveryConfig::class);
+		$this->wcSettingsConfigMock = $this->createMock(WcSettingsConfig::class);
+
+		$this->compoundCarrierFactory = new CompoundCarrierCollectionFactory();
+		$this->vendorCollectionFactory = new VendorCollectionFactory();
+		$this->featureFlagMock = $this->createMock(FeatureFlagManager::class);
+
+		$this->packetaPickupPointsConfig = new PacketaPickupPointsConfig(
+			$this->compoundCarrierFactory,
+			$this->vendorCollectionFactory,
+			$this->featureFlagMock
+		);
+
+			$this->optionsPage = new OptionsPage(
```

Comment:
Rádi bychom v pluginu měli testy stateless, tedy každý test by si měl vytvořit vlastní instanci testované služby.
Až na výjimky by to tedy bylo bez setUp metod a mocků v properties.

Reply:
Rozumím, ale tady mi přišlo lepší při tolika mocích to dát do setUp, když někdo bude psát další test, tak ať se s tím nemusí zdřovat. Ale pravda neprogramujeme pro co by kdyby, takže to dám přímo do toho testu.